### PR TITLE
Merge release/5.1 to develop

### PR DESCRIPTION
**Pull request by rohitsinha54**

### CODE REVIEW

File: cdap-examples/resources/weblog-service.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-deploy app \$CDAP_HOME/examples/CubeService/target/CubeServiceApp-6.0.0-SNAPSHOT.jar
+deploy app \$CDAP_HOME/examples/CubeService/target/CubeServiceApp-5.1.0.jar
```
Comment (rohitsinha54) [inline cdap-examples/resources/weblog-service.txt]: picked one incorrect. FIxing it.

**Comment by rohitsinha54** [comment]
Build: https://builds.cask.co/browse/CDAP-DUT6678

File: cdap-distributions/src/emr/install.sh [hunk 2]
```diff
@@ -24,9 +24,9 @@ CDAP_BRANCH=${CDAP_BRANCH:-release/5.0}
 # Optional tag to checkout - All released versions of this script should set this
 # like this: CDAP_TAG=${CDAP_TAG:+tag} as this allows setting tag to empty/null
 # otherwise, it should be CDAP_TAG=''
-CDAP_TAG=${CDAP_TAG:+v5.1.0}
+CDAP_TAG=${CDAP_TAG:+v5.1.2}
```
Comment (anwar6953) [inline cdap-distributions/src/emr/install.sh]: should this be v6.0.0?

File: cdap-distributions/src/hdinsight/pkg/install.sh [hunk 3]
```diff
@@ -26,7 +26,7 @@ CDAP_BRANCH='release/5.0'
 # otherwise, it should be CDAP_TAG=''
 CDAP_TAG=${CDAP_TAG:+hdi5.0}
 # The CDAP package version passed to Chef
-CDAP_VERSION='5.1.0-1'
+CDAP_VERSION='5.1.2-1'
```
Comment (anwar6953) [inline cdap-distributions/src/hdinsight/pkg/install.sh]: same

File: cdap-distributions/src/emr/install.sh (hunk 2 above)
Comment (rohitsinha54) [reply cdap-distributions/src/emr/install.sh]: @dereklwood : Can you confirm the setting here?

File: cdap-distributions/src/hdinsight/pkg/install.sh (hunk 3 above)
Comment (rohitsinha54) [reply cdap-distributions/src/hdinsight/pkg/install.sh]: @dereklwood : Can you confirm the setting here?

Comment (albertshau) [reply cdap-distributions/src/hdinsight/pkg/install.sh]: Can we be sure to document this, I think it has come up many times

Comment (dereklwood) [reply cdap-distributions/src/hdinsight/pkg/install.sh]: We've been following the simple "set it to the next release" rule.  I assume develop will be version bumped to 5.1.3-SNAPSHOT?  In that case this should be 5.1.3.  Whether it happens now in this merge pr, or in a subsequent version bump PR doesn't matter (but i would suggest version bump PR, keeping this consistent with the rest of the branch).

File: cdap-distributions/src/emr/install.sh (hunk 2 above)
Comment (dereklwood) [reply cdap-distributions/src/emr/install.sh]: We've been following the simple "set it to the next release" rule.  I assume develop will be version bumped to 5.1.3-SNAPSHOT?  In that case this should be 5.1.3.  Whether it happens now in this merge pr, or in a subsequent version bump PR doesn't matter (but i would suggest version bump PR, keeping this consistent with the rest of the branch).

Comment (rohitsinha54) [reply cdap-distributions/src/emr/install.sh]: Develop is on 6.0.0-SNAPSHOT so I will put 6.0.0 here.

File: cdap-distributions/src/hdinsight/pkg/install.sh (hunk 3 above)
Comment (dereklwood) [reply cdap-distributions/src/hdinsight/pkg/install.sh]: nvm, looks like develop branch is already 6.0.0-SNAPSHOT so 6.0.0

File: cdap-distributions/src/emr/install.sh (hunk 2 above)
Comment (dereklwood) [reply cdap-distributions/src/emr/install.sh]: nvm, looks like develop branch is already 6.0.0-SNAPSHOT so 6.0.0

Comment (rohitsinha54) [reply cdap-distributions/src/emr/install.sh]: @dereklwood 

What should be the value above for 
```
CDAP_BRANCH=${CDAP_BRANCH:-release/5.0}
```
This is still at 5.0 should this be develop

Comment (dereklwood) [reply cdap-distributions/src/emr/install.sh]: also update to next release:  `CDAP_BRANCH=${CDAP_BRANCH:-release/6.0}`.  

We should've updated 5.1/5.2 as well, but fortunately shouldn't have any impact as there have been no changes to [cdap-conf.json](https://github.com/cdapio/cdap/blob/684d78015ca5583aff368d54c4cf6a4e43ebc514/cdap-distributions/src/emr/cdap-conf.json)

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by rohitsinha54** [comment]
Waiting on build: https://builds.cask.co/browse/CDAP-DUT6678-3

**Pull request merged**